modules/thoughtleadrBidAdapter.js

```diff
@@ -36,47 +73,50 @@ var ThoughtleadrAdapter = (function () {
 
   ThoughtleadrAdapter.prototype.requestPlacement = function (bid) {
     var _this = this;
-    var rid = utils.generateUUID(null);
+    var uid = getUid();
     var size = ThoughtleadrAdapter.getSizes(bid.sizes);
 
-    window.tldr.requestPrebid(bid.params.placementId, rid).then(function (params) {
-      if (!params || !params.bid) {
-        utils.logError('invalid response from tldr.requestPrebid', undefined, undefined);
-        return;
+    ajax_1.ajax('' + URL_API + bid.params.placementId + '/header-bid.json?uid=' + uid, function (response) {
+      var wonBid = JSON.parse(response);
+      if (wonBid.cookie_syncs) {
+        _this.syncCookies(wonBid.cookie_syncs);
       }
 
-      _this.receiver = function (ev) {
-        if (ev.origin === location.origin &&
-          ev.data && ev.data.TLDR_REQUEST && ev.data.TLDR_REQUEST.rid === rid) {
-          ev.source.postMessage({TLDR_RESPONSE: {config: params.config, rid: rid}}, location.origin);
-          _this.stopListen();
-        }
-      };
-      window.addEventListener('message', _this.receiver, false);
-      setTimeout(function () {
-        return _this.stopListen();
-      }, 5000);
-
       var bidObject;
-      if (params.bid.code === BID_AVAILABLE) {
-        bidObject = bidfactory.createBid(params.bid.code);
+      if (wonBid && wonBid.amount) {
+        bidObject = bidfactory.createBid(BID_AVAILABLE);
         bidObject.bidderCode = 'thoughtleadr';
-        bidObject.cpm = params.bid.cpm;
-        bidObject.ad = params.bid.ad;
+        bidObject.cpm = wonBid.amount;
+        bidObject.ad =
+          '<script src="' + URL_CDN + 'bid.js" header-bid-token="' + wonBid.header_bid_token + '"></script>';
```

Comment:
This is also vulnerable to cross-script injection. Creating a node object and using `setAttribute` makes it safe. (See the same OWASP link I mentioned above)